packages/svelte/src/internal/client/runtime.js

```diff
@@ -653,12 +653,13 @@ function process_effects(effect, filter_flags, shallow, collected_effects) {
 	}
 
 	if (effects.length > 0) {
-		if ((filter_flags & EFFECT) !== 0) {
-			collected_effects.push(...effects);
-		}
-
-		if (!shallow) {
-			for (var i = 0; i < effects.length; i++) {
+		// We might be dealing with many effects here, far more than can be spread into
+		// an array push call (callstack overflow). So let's deal with each effect in a loop.
+		for (var i = 0; i < effects.length; i++) {
+			if ((filter_flags & EFFECT) !== 0) {
```

Comment:
Why is the if condition inside the for loop? It looks like it isn't dependent on the content so can be switched

Reply:
What do you mean? Feel free to put up a PR to show what you mean. :)